tests/copilot_helper/test_command_conduit.py

```diff
@@ -116,52 +114,89 @@ def test_get_connection_secret_arn_when_secret_does_not_exist(mock_application):
         get_connection_secret_arn(mock_application, "development", "POSTGRES")
 
 
+@pytest.mark.parametrize(
+    "addon_type, addon_name",
+    [
+        ("postgres", "custom-name-postgres"),
+        ("postgres", "custom-name-rds-postgres"),
+        ("redis", "custom-name-redis"),
+        ("opensearch", "custom-name-opensearch"),
+    ],
+)
 @patch("subprocess.call")
 @patch("dbt_copilot_helper.commands.conduit.get_connection_secret_arn", return_value="test-arn")
-def test_create_addon_client_task(get_connection_secret_arn, subprocess_call, mock_application):
+def test_create_addon_client_task(
+    get_connection_secret_arn, subprocess_call, addon_type, addon_name, mock_application
+):
     """Test that, given app and environment strings, create_addon_client_task
     calls get_connection_secret_arn with the default secret name and
     subsequently subprocess.call with the correct secret ARN."""
     from dbt_copilot_helper.commands.conduit import create_addon_client_task
 
-    addon_name = "custom-name-postgres"
     task_name = mock_task_name(addon_name)
-    create_addon_client_task(mock_application, "development", "postgres", addon_name, task_name)
+    create_addon_client_task(mock_application, "development", addon_type, addon_name, task_name)
+    secret_name = mock_connection_secret_name(mock_application, addon_type, addon_name)
 
-    get_connection_secret_arn.assert_called_once_with(mock_application, "development", addon_name)
+    get_connection_secret_arn.assert_called_once_with(mock_application, "development", secret_name)
     subprocess_call.assert_called_once_with(
         "copilot task run --app test-application --env development "
         f"--task-group-name {task_name} "
-        "--image public.ecr.aws/uktrade/tunnel:postgres "
+        f"--image public.ecr.aws/uktrade/tunnel:{addon_type} "
         "--secrets CONNECTION_SECRET=test-arn "
         "--platform-os linux "
         "--platform-arch arm64",
         shell=True,
     )
 
 
-@patch("subprocess.call")
-@patch(
-    "dbt_copilot_helper.commands.conduit.get_connection_secret_arn", return_value="test-named-arn"
+@pytest.mark.parametrize(
+    "write, admin",
+    [
+        (False, False),
+        (False, True),
+        (True, False),
+        (True, True),
+    ],
 )
-def test_create_addon_client_task_with_addon_name(
-    get_connection_secret_arn, subprocess_call, mock_application
+@pytest.mark.parametrize(
+    "addon_type, addon_name",
+    [
+        ("postgres", "custom-name-postgres"),
+        ("postgres", "custom-name-rds-postgres"),
+        ("redis", "custom-name-redis"),
+        ("opensearch", "custom-name-opensearch"),
+    ],
+)
```

Comment:
What's happening here with the two `paramerize`s?

Oh wow, today I learnt...

![image](https://github.com/uktrade/copilot-tools/assets/8738245/58fe077b-f373-4e7d-b010-6a5894110292)

Reply:
You can stack multiple parametrize to get combinations of all of the arguments